### ISSUE TITLE
[Android armv8] Application hang on` Sprite` creation on `KeyboardEvent` using `<runtimeInBackgroundThread>true</runtimeInBackgroundThread>` for some devices

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#3023]
## Problem Description
Application hang on `Sprite` creation on `KeyboardEvent` using `<runtimeInBackgroundThread>true</runtimeInBackgroundThread>` for some Android devices with armv8 builds.
Almost all applications could be affected using `<runtimeInBackgroundThread>true</runtimeInBackgroundThread>`.

Tested with latest AIR 50.2.2.3 and AIR 50.2.2.4 versions with multiple different Android devices with different OS versions with different applications and architectures.
100% reproducible with Sony Xperia 5 III, Android 13
With Samsung Galaxy S9, Samsung Galaxy A5 and Samsung Galaxy Tab A8 SM-X200 no such issues.
Issue only reproducible with armv8 builds. No such issue using armv7. Didn' test it with x86 and x86_64.
There is no such issue if you create `Sprite` outside `KeyboardEvent` listener.
There is no such issue without `<runtimeInBackgroundThread>true</runtimeInBackgroundThread>`.

May be related issue:
https://github.com/airsdk/Adobe-Runtime-Support/issues/2888
https://github.com/airsdk/Adobe-Runtime-Support/issues/2326

## Steps to Reproduce
Launch application with code below and press device Back button. It just create new `Sprite`.

Application example with sources attached.
[android_runtimeinbackgroundthread_keyboardevent_sprite_bug.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14037818/android_runtimeinbackgroundthread_keyboardevent_sprite_bug.zip)

```
package {
	import flash.display.Sprite;
	import flash.desktop.NativeApplication;
	import flash.events.KeyboardEvent;
	import flash.events.Event;
	import flash.ui.Keyboard;
	
	public class AndroidRuntimeInBackgroundThreadKeyboardEventSpriteBug extends Sprite {
		private var sprite:Sprite = new Sprite();
		
		public function AndroidRuntimeInBackgroundThreadKeyboardEventSpriteBug() {
			sprite.graphics.beginFill(0xff0000);
			sprite.graphics.drawRect(0, 0, 100, 100);
			sprite.graphics.endFill();
			sprite.x = 100;
			sprite.y = 100;
			addChild(sprite);
			
			addEventListener(Event.ENTER_FRAME, enterFrame);
			
			NativeApplication.nativeApplication.addEventListener(KeyboardEvent.KEY_DOWN, keyDown);
		}
		
		private function keyDown(e:KeyboardEvent):void {
			var keyCode:uint = e.keyCode;
			switch (e.keyCode){
				case Keyboard.BACK:
					trace("keyDown");
					e.preventDefault();
					e.stopPropagation();
					e.stopImmediatePropagation();
					new Sprite();//This line cause hang
					break;
				default:
					break;
			}
		}
	
		public function enterFrame(e:Event):void {
			sprite.rotation += 1;
		}
	}
}
```

**Actual Result:**
All render "stops" and application "hangs". You cannot interract with application anymore.
Scout session stops.
Nothing strange in Logcat.

**Expected Result:**
No application "hangs".

## Known Workarounds
1) Do not use `<runtimeInBackgroundThread>true</runtimeInBackgroundThread>`.
2) Use `setTimeout` when you need to create `Sprite` in `KeyboardEvent` listener.

**Comment by bobaoapae** [comment]
This flash has a lot issues yet, it's not usable in production. I get worse performance using this than without this. Setting to true reduce the ANRs due to fake behavior of responsive application, but the application it's self run slowly..

**Comment by ajwfrost** [comment]
Yes, I think there are a whole load of issues caused by the interactivity between Android and AIR when we try to put the AIR runtime into a background thread. So for this instance, there is a user hitting a key, so we are receiving a UI event from Android, but would then need to send that into AIR asynchronously .. and that means we lose the option to prevent default behaviour etc - we can't wait for the AIR thread to complete as that would negate the whole idea of this mechanism. Currently of course, we seem to end up in scenarios where a UI thread has triggered some async behaviour but the async behaviour then needs to trigger something back on the UI thread, which is I think where a lot of these issues are coming from.

TBH the whole `runtimeInBackgroundThread` thing is probably not a viable option... the better approach may be just keeping everything "normal" in the main UI thread, but then seeing what specific AS3 behaviours are causing ANRs, and looking instead to optimise those...

**Comment by itlancer** [comment]
@ajwfrost 
In my cases `runtimeInBackgroundThread` could really increase performance when application use camera.
Without it AIR for Android have really terrible performace with camera usage: https://github.com/airsdk/Adobe-Runtime-Support/issues/1980
